test/testcafe/framework/page-objects/ChallengeFactorPageObject.js

```diff
@@ -71,11 +79,15 @@ export default class ChallengeFactorPageObject extends BasePageObject {
     return this.form.getTextBoxErrorMessage(PASSWORD_FIELD);
   }
 
-  async clickSendAgainLink() {
-    await this.form.clickElement('.resend-email-view a.resend-link');
+  async clickSendAgainLink(resendContainer = '.resend-email-view') {
+    await this.form.clickElement(`${resendContainer} a.resend-link`);
+  }
+
+  getSaveButtonLabel(saveButtonSelector = '.button-primary') {
+    return this.form.getElement(saveButtonSelector).value;
   }
 
-  getSaveButtonLabel() {
-    return this.form.getElement('.button-primary').value;
+  elementHasClass(elementSelector, className) {
```

Comment:
better to extend `ChallengeFactorPageObject` to `ChallengePhonePageObject` and adds your page specify functions.